quinn/src/streams.rs

```diff
@@ -353,42 +353,38 @@ where
         buf: &mut ReadBuf<'_>,
     ) -> Poll<Result<(), ReadError>> {
         self.poll_read_generic(cx, |conn, stream| {
-            conn.inner
-                .read(stream, buf.remaining())
-                .map(|val| val.map(|chunk| buf.put_slice(&chunk)))
+            let mut chunks = conn.inner.read(stream, true)?;
+            while dbg!(buf.remaining()) > 0 {
```

Comment:
Remove `dbg!` before shipping this. Good idea is also to run the benchmark, and see if there is any regression (there will definitely be with extra output).

Reply:
Yeah, will of course remove that.